ext4: Fix update of mtime and ctime on rename

From: Jan Kara <jack@suse.cz>

The patch below makes ext4 update mtime and ctime of the directory
into which we move file even if the directory entry already exists.

Signed-off-by: Jan Kara <jack@suse.cz>
Signed-off-by: Mingming Cao <cmm@us.ibm.com>
---
 fs/ext4/namei.c |    3 +++
 1 file changed, 3 insertions(+)

Index: linux-2.6.25-rc9/fs/ext4/namei.c
===================================================================
--- linux-2.6.25-rc9.orig/fs/ext4/namei.c	2008-04-14 10:25:34.000000000 -0700
+++ linux-2.6.25-rc9/fs/ext4/namei.c	2008-04-14 10:25:38.000000000 -0700
@@ -2354,6 +2354,9 @@ static int ext4_rename (struct inode * o
 					      EXT4_FEATURE_INCOMPAT_FILETYPE))
 			new_de->file_type = old_de->file_type;
 		new_dir->i_version++;
+		new_dir->i_ctime = new_dir->i_mtime =
+					ext4_current_time(new_dir);
+		ext4_mark_inode_dirty(handle, new_dir);
 		BUFFER_TRACE(new_bh, "call ext4_journal_dirty_metadata");
 		ext4_journal_dirty_metadata(handle, new_bh);
 		brelse(new_bh);
